lib/multiplexer/proxyline.go

```diff
@@ -33,27 +35,50 @@ import (
 	"github.com/gravitational/trace"
 )
 
+// PP2Type is the PROXY protocol v2 TLV type
+type PP2Type byte
+
 const (
 	// TCP4 is TCP over IPv4
 	TCP4 = "TCP4"
 	// TCP6 is tCP over IPv6
 	TCP6 = "TCP6"
 	// Unknown is unsupported or unknown protocol
 	UNKNOWN = "UNKNOWN"
+
+	PP2TypeNOOP PP2Type = 0x04 // No-op used for padding
+
+	// Known custom types, spec allows to use 0xE0 - 0xEF for custom types
+	PP2TypeGCP   PP2Type = 0xE0 // https://cloud.google.com/vpc/docs/configure-private-service-connect-producer
```

Comment:
Does it make sense to add this types? Will they be used? Maybe we can add them then?

Reply:
It's there just so we don't accidentally don't create type that shadows those, to signify it's already taken. It was also a suggestion from Alan Parra.